python/ray/util/dask/__init__.py

```diff
@@ -6,10 +7,43 @@
 )
 from .optimizations import dataframe_optimize
 
+dask_persist = dask.persist
+
+
+def ray_dask_persist(*args, **kwargs):
+    kwargs["ray_persist"] = True
+    return dask_persist(*args, **kwargs)
+
+
+ray_dask_persist.__doc__ = dask_persist.__doc__
+
+dask_persist_mixin = dask.base.DaskMethodsMixin.persist
+
+
+def ray_dask_persist_mixin(self, **kwargs):
+    kwargs["ray_persist"] = True
+    return dask_persist_mixin(self, **kwargs)
+
+
+ray_dask_persist_mixin.__doc__ = dask_persist_mixin.__doc__
+
+
+# We patch dask in order to inject a kwarg into its `dask.persist()` calls,
+# which the Dask-on-Ray scheduler needs.
+# FIXME(Clark): Monkey patching is bad and we should try to avoid this.
+def patch_dask(ray_dask_persist, ray_dask_persist_mixin):
+    dask.persist = ray_dask_persist
+    dask.base.DaskMethodsMixin.persist = ray_dask_persist_mixin
+
+
+patch_dask(ray_dask_persist, ray_dask_persist_mixin)
+
 __all__ = [
     # Schedulers
     "ray_dask_get",
     "ray_dask_get_sync",
+    # Helpers
+    "ray_dask_persist",
```

Comment:
We still export `ray_dask_persist` in case users want to explicitly use it in their code.

Reply:
If I understand correctly, this is the same as the normal `dask.persist` call? Should we call this something else? The name is a bit confusing because it has `ray` in it.

Reply:
It injects a `ray_persist` kwarg that only the Dask-on-Ray schedulers understand, and it's supposed to be a `dask.persist()` method tailored for Dask-on-Ray, so I thought that matching the scheduler naming scheme of `ray_dask_{}` would make its semantics more clear. Although I'm tempted to rename all of these to `get`, `get_sync`, and `persist`, with the context of them importing from `ray.util.dask` implying their semantics... But that would be a breaking API change.